okhttp/src/main/java/okhttp3/Address.java

```diff
@@ -195,4 +207,25 @@ public CertificatePinner certificatePinner() {
     result.append("}");
     return result.toString();
   }
+
+  public boolean isCoalescable() {
+    return url().isHttps()
+        && this.proxy() == null
+        && hostnameVerifier == OkHostnameVerifier.INSTANCE;
+  }
+
+  public List<InetAddress> dnsLookup(String hostname) throws UnknownHostException {
```

Comment:
In particular, why isn’t `hostname` just the hostname of this address? That’s surprising